### ISSUE TITLE
Cannot Reproduce Drone → Satellite Results on University-1652 Dataset

### ISSUE THREAD

**Issue by ckdghks** [Skyy93/Sample4Geo#41]
Hi, thanks for the great work.

I'm currently trying to reproduce the Drone → Satellite results on the University-1652 dataset using your Sample4Geo codebase. However, I’m not able to reach the reported performance, even when closely following the settings.

Here is a summary of my current setup:

@dataclass
class Configuration:
    model: str = 'convnext_base.fb_in22k_ft_in1k_384'
    img_size: int = 384
    mixed_precision: bool = True
    custom_sampling: bool = True
    seed = 1
    epochs: int = 40
    batch_size: int = 16
    verbose: bool = True
    gpu_ids: tuple = (0, 1)

    batch_size_eval: int = 128
    eval_every_n_epoch: int = 1
    normalize_features: bool = True
    eval_gallery_n: int = -1

    clip_grad = 100.
    decay_exclue_bias: bool = False
    grad_checkpointing: bool = False
    label_smoothing: float = 0.1

    lr: float = 0.001
    scheduler: str = "cosine"
    warmup_epochs: int = 0.1
    lr_end: float = 0.0001

    dataset: str = 'U1652-D2S'
    data_folder: str = "/workspace/mount/SSD_2T_a/AAM_Data/University_dataset/University-Release"

    prob_flip: float = 0.5
    model_path: str = "./university_original"
    zero_shot: bool = False
    checkpoint_start = None
    num_workers: int = 2
    device: str = 'cuda'
    cudnn_benchmark: bool = True
    cudnn_deterministic: bool = False

📊 Evaluation Stats
Model: convnext_base.fb_in22k_ft_in1k_384

GPUs: 2

Image Size: (384, 384)

Normalization: mean = (0.485, 0.456, 0.406), std = (0.229, 0.224, 0.225)

Query images (Drone): 37,855

Gallery images (Satellite): 951

📉 Evaluation Result
Recall@1: 81.73
Recall@5: 94.33
Recall@10: 96.35
Recall@top1: 96.55
mAP: 84.58
(I also tried increasing the batch size to 128 (as InfoNCE performance is often sensitive to batch size), but the result did not improve significantly.)

❗️Issue:
Even with the above configuration and model, I am unable to reproduce the performance reported in the paper. If I use the pretrained model you provided, the performance exceeds 80% Recall@1 in just 1 epoch. However, training from scratch does not reproduce the paper results, even after 40 epochs.

Could you clarify:

Was the reported performance obtained using a pretrained model?

Are there any hidden tricks (e.g., sampling ratio, specific data filtering, loss weight tuning)?

Any suggestion for achieving the reported results starting from scratch?

Thanks in advance!

**Comment by Skyy93** [comment]
Thank you for using and reproducing our work, for University-1652 we only use 1 Epoch, otherwise the network overfits.

**Comment by Skyy93** [comment]
Ah also i see you used batch size 16, you should use 128 otherwise the results are worse, this is mostly due to the InfoNCE, Higher batch size=better generalization. 

Perhaps you could solve this by gradient accumulation.

**Comment by wincj** [comment]
> Hi, thanks for the great work.
> 
> I'm currently trying to reproduce the Drone → Satellite results on the University-1652 dataset using your Sample4Geo codebase. However, I’m not able to reach the reported performance, even when closely following the settings.
> 
> Here is a summary of my current setup:
> 
> [@DataClass](https://github.com/DataClass) class Configuration: model: str = 'convnext_base.fb_in22k_ft_in1k_384' img_size: int = 384 mixed_precision: bool = True custom_sampling: bool = True seed = 1 epochs: int = 40 batch_size: int = 16 verbose: bool = True gpu_ids: tuple = (0, 1)
> 
> ```
> batch_size_eval: int = 128
> eval_every_n_epoch: int = 1
> normalize_features: bool = True
> eval_gallery_n: int = -1
> 
> clip_grad = 100.
> decay_exclue_bias: bool = False
> grad_checkpointing: bool = False
> label_smoothing: float = 0.1
> 
> lr: float = 0.001
> scheduler: str = "cosine"
> warmup_epochs: int = 0.1
> lr_end: float = 0.0001
> 
> dataset: str = 'U1652-D2S'
> data_folder: str = "/workspace/mount/SSD_2T_a/AAM_Data/University_dataset/University-Release"
> 
> prob_flip: float = 0.5
> model_path: str = "./university_original"
> zero_shot: bool = False
> checkpoint_start = None
> num_workers: int = 2
> device: str = 'cuda'
> cudnn_benchmark: bool = True
> cudnn_deterministic: bool = False
> ```
> 
> 📊 Evaluation Stats Model: convnext_base.fb_in22k_ft_in1k_384
> 
> GPUs: 2
> 
> Image Size: (384, 384)
> 
> Normalization: mean = (0.485, 0.456, 0.406), std = (0.229, 0.224, 0.225)
> 
> Query images (Drone): 37,855
> 
> Gallery images (Satellite): 951
> 
> 📉 Evaluation Result Recall@1: 81.73 Recall@5: 94.33 Recall@10: 96.35 Recall@top1: 96.55 mAP: 84.58 (I also tried increasing the batch size to 128 (as InfoNCE performance is often sensitive to batch size), but the result did not improve significantly.)
> 
> ❗️Issue: Even with the above configuration and model, I am unable to reproduce the performance reported in the paper. If I use the pretrained model you provided, the performance exceeds 80% Recall@1 in just 1 epoch. However, training from scratch does not reproduce the paper results, even after 40 epochs.
> 
> Could you clarify:
> 
> Was the reported performance obtained using a pretrained model?
> 
> Are there any hidden tricks (e.g., sampling ratio, specific data filtering, loss weight tuning)?
> 
> Any suggestion for achieving the reported results starting from scratch?
> 
> Thanks in advance!

Hello, may I ask if I have obtained the raw datasets of CUSUA and VIGOR, but it seems that they cannot be trained directly and need to be formatted? How did you format the dataset to adapt the code for training

**Comment by Skyy93** [comment]
We usually did no special formatting to prevent exactly this issue, what kind of issue are you running into?
